src/main/java/Main.java

```diff
@@ -27,20 +24,26 @@ public static Session getSession() throws HibernateException {
         return ourSessionFactory.openSession();
     }
 
-    public static void main(final String[] args) throws Exception {
+    public static void main(final String[] args) {
         final Session session = getSession();
         try {
-            System.out.println("querying all the managed entities...");
+            log.info("querying all the managed entities...");
             final Metamodel metamodel = session.getSessionFactory().getMetamodel();
             for (EntityType<?> entityType : metamodel.getEntities()) {
                 final String entityName = entityType.getName();
                 final Query query = session.createQuery("from " + entityName);
-                System.out.println("executing: " + query.getQueryString());
+                log.info("executing: " + query.getQueryString());
                 for (Object o : query.list()) {
-                    System.out.println("  " + o);
+                    log.info("  " + o);
                 }
             }
-        } finally {
+        } catch(QueryException e) {
+            log.error("Query Error", e);
+        }
+        catch(SessionException e) {
+            log.error("Session Error", e);
+        }
+        finally {
```

Comment:
I added two catch statements that are specific to the possible errors that can happen here. This is why I use a QueryException and SessionException and log these when they happen. Query and session exceptions are within the Hibernate exceptions import so they can be used from just throwing a hibernate exception. However, a hibernate exception was only used with getSession() and not main.

Reply:
Both catches are doing the same thing.  Multiple catches are good if they are performing different behavior. If they are performing the same behavior, it's analogous to having an if test where every test gives you the same outcome.